appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -0,0 +1,201 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.appinventor.client.editor.simple.palette.SimplePaletteItem;
+import com.google.appinventor.client.widgets.dnd.DragSource;
+import com.google.appinventor.components.common.ComponentConstants;
+import com.google.gwt.event.logical.shared.AttachEvent;
+import org.pepstock.charba.client.resources.EmbeddedResources;
+import org.pepstock.charba.client.resources.ResourcesType;
+
+public final class MockChart extends MockContainer {
+    public static final String TYPE = "Chart";
+
+    private static final String PROPERTY_NAME_TYPE = "Type";
+    private static final String PROPERTY_NAME_DESCRIPTION = "Description";
+
+    static {
+        ResourcesType.setClientBundle(EmbeddedResources.INSTANCE);
+    }
+
+    protected MockChartView chartView;
+
+    // Legal values for type are defined in
+    // com.google.appinventor.components.runtime.Component.java.
+    private int type;
+
+    /**
+     * Creates a new instance of a visible component.
+     *
+     * @param editor editor of source file the component belongs to
+     */
+    public MockChart(SimpleEditor editor) {
+        super(editor, TYPE, images.image(), new MockChartLayout());
+
+        // Since the Mcok Chart component is not a container in a normal
+        // sense (attached components should not be visible), the Chart Widget
+        // is added to the root panel, and the root panel itself is initialized.
+        // This is done to ensure that Mock Chart Data components can be dragged
+        // onto the Chart itself, rather than outside the Chart component.
+        rootPanel.setStylePrimaryName("ode-SimpleMockComponent");
+
+        // Since default type property does not invoke the setter,
+        // initially, the Chart's type setter should be invoked
+        // with the default value.
+        setTypeProperty("0");
+
+        initComponent(rootPanel);
```

Comment:
Is it fine to initialize the rootPanel as the component? I found this to work quite nicely, especially since there is only one Component at a time in the Container, and all the attached components are not actually visible.

Reply:
Also see line 50, where I use the addAttachHandler on the rootPanel. Is this fine?